src/ASM/ASMs2DLag.C

```diff
@@ -928,8 +928,7 @@ IntMat ASMs2DLag::getElmNodes (int basis) const
         nx[d] += srf->basis(d).order()-1;
       uprev = *uit;
     }
-    if (srf->basis(d).order() > 2)
-      nx[d]++;
+    nx[d]++;
```

Comment:
If this fix is right (an I assume you have indications it is), it is better do initialize `nx` to 1 (i.e., change L919 to `int nx[2] = { 1, 1 };`) and we can skip the increment totally.

Reply:
yes, extensively tested. changed to init to 1